File: Java/simple/WhileNullPointerBug/src/at/faultycode/java/simple/WhileNullPointerBug.java
package at.faultycode.java.simple;

import java.util.logging.Logger;

/**
 * @author MaAb
 */
public class WhileNullPointerBug {

	/** Default Java logger **/
	private static final Logger LOGGER = Logger
			.getLogger(WhileNullPointerBug.class.getName());

	private WhileNullPointerBug() {}
	
	/**
	 * Entry point of program
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		String message = "No arguments are specified!";

		int i = 0;
		while (i < args.length) {
			message = null;
			i++;
		}

		if (!message.isEmpty()) {
			LOGGER.info(message);
		}
	}
}

File: C++/simple/WhileNullPointerBug/src/WhileNullPointerBug.cpp
///
/// Author: MaAb
///

struct A {
   int num;
};

///
/// Entry point of program
///
int main(int argc, char** argv) {
    A *a = A();
    *a->num = argc;
    
    int i = 0;
    while (i < argc) {
      *a = nullptr;
    }
    
    return a->num;
}
